geoopt/manifolds/poincare/math.py

```diff
@@ -528,26 +532,26 @@ def clip_tangent(x, u, *, c=1.0, dim=-1):
         ball negative curvature
     dim : int
         reduction dimension to compute norm
+    eps : float
+        stability parameter
 
     Returns
     -------
     tensor
         same tangent vector with reasonable values
     """
-    return _clip_tangent(x, u, c, dim=dim)
+    return _clip_tangent(x, u, c, dim=dim, eps=eps)
 
 
-def _clip_tangent(x, u, c, dim: int = -1):
+def _clip_tangent(x, u, c, dim: int = -1, eps=None):
     # get the almost infinite vecotor estimate
     # this is the norm of travel vector to the opposite pole
     s = x.size(dim)
     p = torch.ones((s,), dtype=x.dtype, device=x.device)
     p = p / s ** 0.5 / (c ** 0.5)
-    p = _project(p, c, dim=dim)
+    p = _project(p, c, eps=eps)
```

Comment:
You sure you wanted to remove `dim=dim`?

Reply:
This is intended here